### DISCUSSION TITLE
Scope the dependency to a navigation graph

### DISCUSSION THREAD

**Discussion by BenderNK** [pointfreeco/swift-dependencies#294] [General]
I created a dependency called `LocationManager` by following the documentation in this library. It works great - no problems there. However, I would like to scope this dependency to a part of the navigation graph so that when I go back to a place where I don't need the said location manager anymore, i would like to release it from memory. I looked around but I couldn't find a way to configure the dependency this way. If there is already one and I missed that, apologies in advance. I took a snapshot of my memory graph where I would like no longer see `LocationManager`. However, CachedValues object still retains a reference to this dependency. Is this feasible? Check out the following screenshot. 

<img width="681" alt="Screenshot 2024-10-29 at 2 21 06 PM" src="https://github.com/user-attachments/assets/8478d2ba-9640-4757-a0da-b82378632fec">

**Comment by rhysm94** [comment]
This can work, but you need to "bake in" the ability to create and destroy the dependency within your own code.
Maybe you could keep a `NoOpLocationManager`/`UnimplementedLocationManager` as the default value in a `LockIsolated`, and then call a method on your dependency like this:

```swift
@Dependency(\.locationManager) var locationManager
locationManager.create()
```

where `create` (and `destroy`) looks _something_ like this:

```swift
static var liveValue: Self {
  let locationManager = LockIsolated<LiveLocationManager>(.noOp)

  return Self(
    create: {
      locationManager.withValue { manager in
        manager = .live
      }
    },
    destroy: {
      locationManager.withValue { manager in
        manager = .noOp
      }
    }
```

**Comment by BenderNK** [comment]
Thanks for the advice. I will implement this in our project and let you know - about a week or so from now.
